test/components/views/beacon/BeaconMarker-test.tsx

```diff
@@ -93,38 +99,46 @@ describe("<BeaconMarker />", () => {
     it("renders nothing when beacon is not live", () => {
         const room = setupRoom([notLiveEvent]);
         const beacon = room.currentState.beacons.get(getBeaconInfoIdentifier(notLiveEvent));
-        const component = getComponent({ beacon });
-        expect(component.html()).toBe(null);
+        const { container } = renderComponent({ beacon });
+        expect(container.innerHTML).toBe("");
     });
 
     it("renders nothing when beacon has no location", () => {
         const room = setupRoom([defaultEvent]);
         const beacon = room.currentState.beacons.get(getBeaconInfoIdentifier(defaultEvent));
-        const component = getComponent({ beacon });
-        expect(component.html()).toBe(null);
+        const { container } = renderComponent({ beacon });
+        expect(container.innerHTML).toBe("");
     });
 
     it("renders marker when beacon has location", () => {
         const room = setupRoom([defaultEvent]);
         const beacon = room.currentState.beacons.get(getBeaconInfoIdentifier(defaultEvent));
-        beacon.addLocations([location1]);
-        const component = getComponent({ beacon });
-        expect(component).toMatchSnapshot();
+        beacon?.addLocations([location1]);
+        const { asFragment } = renderComponent({ beacon });
+        expect(asFragment()).toMatchSnapshot();
     });
 
     it("updates with new locations", () => {
+        // we need to mock the SmartMarker here to be able to check the geoUris
+        const uriSmartMarker = jest.spyOn(mockSmartMarker, "default");
+        uriSmartMarker.mockImplementation((props) => <div data-testid={props.geoUri} />);
```

Comment:
Could you spy on `maplibregl.Marker.setLngLat` to assert the uri is updated instead of this? Similar to how the `SmartMarker` tests work?

Reply:
I'll have a look at that, thanks for the suggestion. It didn't feel good doing this and I'm in a similar scenario trying to rewrite another test